pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -85,7 +85,16 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(env model.Environmen
 	for _, service := range services {
 		config := env.DestinationRule(service.Hostname)
 		for _, port := range service.Ports {
-			hosts := buildClusterHosts(env, service, port)
+			var hosts []*core.Address
+			if proxy.Type == model.Router && service.MeshExternal {
+				// In case of Router (e.g. ingress/egress gateway) and external service, we want the service resolution
+				// to be DNS. The router can't use ORIGINAL_DST since, as opposed to a sidecar, it does not get the
+				// traffic by iptables so original_dst is not set.
+				service.Resolution = model.DNSLB
```

Comment:
see my explanation in the PR. Also, we could have resolution be STATIC and thats perfectly valid in gateway